cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -186,13 +216,18 @@ private AccessToken getNewAccessToken(ConnectionConfig connectionInfo, PrintStre
         credentialValue = reader.readLine(prompt);
       }
       properties.put(credential.getName(), credentialValue);
+      if (credential.getName().contains("username")) {
+        username = credentialValue;
+        connectionInfo = new AuthenticatedConnectionConfig(connectionInfo, username);
+        clientConfig.setConnectionConfig(connectionInfo);
```

Comment:
Again, connectionConfig won't be changed in the parent function unless I set it here.